### ISSUE TITLE
nan in training

### ISSUE THREAD

**Issue by liz6688** [NVlabs/wetectron#12]
Thanks for your code.
There is an nan during training. Can you give some suggestions to solve this problem? 
I set IMS_PER_BATCH = 1,  ITER_SIZE = 1, and keep other parameters unchanged.

**Comment by jason718** [comment]
IMS_PER_BATCH is too small. The default setting is for `IMS_PER_BATCH=8` or `16`. We didn't test the performance under batch_size 1. If you want to do that, you also need to change the other setting accordingly (e.g., reduce the learning rate, train longer, erc)

**Comment by liz6688** [comment]
I aslo trained a model with IMS_PER_BATCH=4. It appeared “WARNING:root:NaN or Inf found in input tensor”. Do you have any suggestions?

**Comment by jason718** [comment]
I would use smaller learning rate, longer warm-up, ITER_SIZE flag.

**Comment by liz6688** [comment]
Thanks. I will try. How small does the learning rate become? Is the drop in learning rate same as the IMS_PER_BATCH, and set SOLVER.BASE_LR= 0.01/8, SOLVER_CDB.BASE_LR=0.001/8?

**Comment by jason718** [comment]
According to some published work, linear scaling seems to be the rule. But I guess it won't be optimal and the performance might be greatly harmed. Maintain a decent batch size is usually beneficial.

**Comment by bradezard131** [comment]
You might try using a learning rate of 0.001 with batch size 2 or 4, similar to OICR and PCL (Ideally you would have the pretrained layers at 0.001 and the new layers at 0.01 but I don't think that is currently supported).

**Comment by jason718** [comment]
closing this for now, feel free to reopen.

**Comment by Fly-dream12** [comment]
I have trained this model on my dataset. The learing rate is set as 0.00025 with batch_size 2 trained on a single GPU,  the base_lr of CDB is set as 0.0001. I trained the model for 15000 iterations for total, however, the loss seems not to be converged and it begins with 1.5 and ends with 1.1.  The evaluation results are also worse,  may be the parameters are wrong. Could you give some suggestions?@jason718

**Comment by jason718** [comment]
> I have trained this model on my dataset. The learing rate is set as 0.00025 with batch_size 2 trained on a single GPU, the base_lr of CDB is set as 0.0001. I trained the model for 15000 iterations for total, however, the loss seems not to be converged and it begins with 1.5 and ends with 1.1. The evaluation results are also worse, may be the parameters are wrong. Could you give some suggestions?@jason718

1. how big is your dataset? how many classes and how many training images?
2. the issue might be the single-GPU. We recommend using 8 GPUs. 
3. try a model w/o CDB first.
4. sometimes MIL loss can be mis-leading. Try to watch the accuracy also.
